appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt

```diff
@@ -1,21 +1,110 @@
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.apps_discovery_activity)
-        Log.d("###", "pkg -- ${applicationContext.packageName} ")
     }
 
     companion object {
         fun getIntent(context: Context) = Intent(context, AppsDiscoveryActivity::class.java)
     }
+
+    private var isBound = false
+    private var transferService: SendKinServiceBase? = null
+    private val connection = object : ServiceConnection {
+
+        override fun onServiceConnected(className: ComponentName, service: IBinder) {
+            val binder = service as SendKinServiceBase.KinTransferServiceBinder
+            transferService = binder.service
+            isBound = true
+            //TODO call only when try to send
+            startSendKinRequest()
+        }
+
+        override fun onServiceDisconnected(arg0: ComponentName) {
+            isBound = false
+        }
+    }
+
+    fun startSendKinRequest() {
+        //TODO
+        //update the transaction bar of the start of the transaction
+        if (isBound) {
+            val thread = Thread(Runnable {
+                val receiver = "SwellyAddress"
+                val amount = 3
+                val memo = "lean"
+
+                try {
+                    val currentBalance = transferService?.currentBalance
+                    Log.d("####", "#### current balanbe is $currentBalance")
+                } catch (balanceException: SendKinServiceBase.BalanceException) {
+                    //TODO need to handle??
+                    Log.d("####", "#### balanceException ${balanceException.message}")
+                }
+
+                try {
+                    val kinTransferComplete = transferService?.transferKin(receiver, amount, memo)!!
+                    //TODO notify the transaction bar of complete
+                    //notify the receiver of the transaction
+                    ReceiveKinServiceBase.notifyTransactionCompleted(baseContext, kinTransferComplete.senderAddress, receiver, amount, kinTransferComplete.transactionId, memo)
+                } catch (kinTransferException: SendKinServiceBase.KinTransferException) {
+                    //TODO notify the transaction bar of error
+                    //notify the receiver of the error
+                    Log.d("####", "#### kinTransferException ${kinTransferException.message}")
+                    ReceiveKinServiceBase.notifyTransactionFailed(baseContext, kinTransferException.toString(), kinTransferException.senderAddress, receiver, amount, memo)
+                }
+            })
+            thread.start()
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        testKinSendService()
+    }
+
+    private fun testKinSendService(){
+        val intent = Intent()
+        //TODO change to receiver pkg
+        val receiverPackageName = applicationContext.packageName
```

Comment:
should be the sender, i.e. this process package name

Reply:
it will be changed when implemented